api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -179,6 +194,22 @@ message DeltaDiscoveryRequest {
   // A list of Resource names to remove from the list of tracked resources.
   repeated string resource_names_unsubscribe = 4;
 
+  // [#not-implemented-hide:]
+  // Alternative to resource_names_subscribe field that allows specifying cache
+  // keys along with each resource name. If this is populated in the
+  // first request for a resource type on a stream, resource_names_subscribe
+  // and resource_names_unsubscribe are ignored for all subsequent requests
+  // for that resource type on that stream.
```

Comment:
In what situation would a client initially send `resources_names_...` and then later switch to `resource_locators_...`?  I think in general, it will be the client's config that tells it which one to use, and that config is not going to change after the client starts up.